clients/ui/frontend/src/components/ToastNotification.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Alert, AlertActionCloseButton, AlertVariant } from '@patternfly/react-core';
+import { Notification } from '~/app/context/NotificationContext';
+import { asEnumMember } from '~/utilities/utils';
+import { useNotification } from '~/app/hooks/useNotification';
+
+const TOAST_NOTIFICATION_TIMEOUT = 8 * 1000;
+
+interface ToastNotificationProps {
+  notification: Notification;
+}
+
+const ToastNotification: React.FC<ToastNotificationProps> = ({ notification }) => {
+  const notifications = useNotification();
+  const [timedOut, setTimedOut] = React.useState(false);
+  const [mouseOver, setMouseOver] = React.useState(false);
+
+  React.useEffect(() => {
+    const handle = setTimeout(() => {
+      setTimedOut(true);
+    }, TOAST_NOTIFICATION_TIMEOUT);
+    return () => {
+      clearTimeout(handle);
+    };
+  }, [setTimedOut]);
+
+  React.useEffect(() => {
+    if (!notification.hidden && timedOut && !mouseOver) {
+      notifications.remove(notification.id)
+    }
+  }, [mouseOver, notification, timedOut]);
+
+  if (notification.hidden) {
+    return null;
+  }
+
+  return (
+    <Alert
+      variant={asEnumMember(notification.status, AlertVariant) ?? undefined}
+      title={notification.title}
+      actionClose={
+        <AlertActionCloseButton onClose={() => notifications.remove(notification.id)} />
+      }
+      onMouseEnter={() => setMouseOver(true)}
+      onMouseLeave={() => setMouseOver(false)}
+    >
+      {notification.message}
```

Comment:
See the description: this is not being passed along properly.

Reply:
Left a comment with the fix